### PR TITLE
Added protection for logger for BackgroudJob. Fixes#23

### PR DESCRIPTION
**Pull request by Trial97**

### CODE REVIEW

File: fsock.go [hunk 1]
```diff
@@ -468,7 +468,9 @@ func (self *FSock) doBackgroudJob(event string) { // add mutex protection
 	evMap := EventToMap(event)
 	jobUuid, has := evMap["Job-UUID"]
 	if !has {
```
Comment (danbogos) [inline fsock.go]: Since it is only one condition, please add the logger check inside the first if.

Comment (Trial97) [reply fsock.go]: We still want to exit function when we can't find the `Job-UUID` even if the logger is nil so we can't put both conditions in the same if only if we check again for has like so:
```
if !has && self.logger != nil {
    self.logger.Err("<FSock> BACKGROUND_JOB with no Job-UUID")
    return
}else if !has{
    return
}
```
It is still needed to added like that?

Comment (danbogos) [reply fsock.go]: OK, did not look it from that point of view. You are right, let's leave it like originally done.

Thanks!

**Comment by danbogos** [comment]
Thanks!

**Pull request merged**